leash.go

```diff
@@ -100,23 +109,39 @@ func run(options GlobalOptions) {
 
 		// start a goroutine that reads from responses and logs.
 		responses := libhoney.Responses()
-		go handleResponses(responses, toBeResent, delaySending, options)
+		stats := newResponseStats()
+		go handleResponses(responses, toBeResent, delaySending, stats, options)
 
 		parsersWG.Add(1)
-		go func(plines chan string) {
+		go func(pTailer *tail.Tailer) {
 			// ProcessLines won't return until lines is closed
-			parser.ProcessLines(plines, toBeSent)
+			parser.ProcessLines(pTailer.Lines, toBeSent)
 			// trigger the sending goroutine to finish up
 			close(toBeSent)
 			// wait for all the events in toBeSent to be handed to libhoney
 			<-doneSending
 			parsersWG.Done()
-		}(lines)
+		}(tailer)
+		hny := &hny{
+			parser: parser,
+			tailer: tailer,
+			rStats: stats,
+		}
+		hnys = append(hnys, hny)
+		go logStats(hny, options.StatusInterval)
```

Comment:
also, let's sanity check that the `tailer` logging does the right thing when multiple files are specified.